src/tree_cmd.cpp

```diff
@@ -549,9 +608,28 @@ static void DrawTile_Trees(TileInfo *ti)
 
 	/* put the trees to draw in a list */
 	uint trees = GetTreeCount(ti->tile);
+	uint growth = GetTreeGrowth(ti->tile);
+	if (growth == TREE_GROWTH_PROCEDURAL) {
+		uint64_t counter = 0;
+
+		/* Use a fixed value for the counter (0) when trees aren't growing. */
+		if (_settings_game.construction.extra_tree_placement != ETP_NO_GROWTH_NO_SPREAD) {
+			uint64_t cycle = tile_hash + (TimerGameTick::counter >> 8);
+			counter = cycle >> 4;
+
+			/* It may happen that counter has not increased yet but the tile was already
+			 * iterated with the new value so we need to adjust.
+			 */
+			if ((cycle & 15) == 15 && WasTileIteratedThisCycle(ti->tile)) counter++;
```

Comment:
What's the 15? Could it be a named constant? (even if just locally)

Reply:
To me this immediately pops out as "test whether all four lower bits are set", maybe it would be easier to recognize with the constants written in hex or binary?
`(cycle & 0b1111) == 0b1111`